src/components/course/routes/ExternalCourseEnrollment.jsx

```diff
@@ -27,7 +27,8 @@ const ExternalCourseEnrollment = () => {
   const { externalCourseFormSubmissionError } = useContext(CourseContext);
   const { data: enterpriseCustomer } = useEnterpriseCustomer();
   const isCourseAssigned = useIsCourseAssigned();
-  const { data: minimalCourseMetadata } = useMinimalCourseMetadata();
+  const { courseRunKey } = useParams();
+  const { data: minimalCourseMetadata } = useMinimalCourseMetadata({courseRunKey});
```

Comment:
Nit: to stick the new paradigm of minimizing as many function args, could you call useParams directly within useMinimalCourseMetadata to avoid needing to pass it as a function arg?